src/MICore/Transports/LocalLinuxTransport.cs

```diff
@@ -187,9 +195,29 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
             reader = new StreamReader(gdbStdOutStream);
         }
 
+        private void FifoWatcher_Deleted(object sender, FileSystemEventArgs e)
+        {
+            if (e.FullPath == _gdbStdInName || e.FullPath == _gdbStdOutName)
+            {
+                Logger?.WriteLine("Gdb fifo deleted, stop debugging");
+                _fifoWatcher.Deleted -= FifoWatcher_Deleted;
+                this.Callback.OnDebuggerProcessExit(null);
+            }
+        }
+
         protected override string GetThreadName()
         {
             return "MI.LocalLinuxTransport";
         }
+
+        public override void Close()
+        {
+            base.Close();
```

Comment:
Jackson's question seems disappear after I rebase the branch, the question is "Since the fifos are always deleted on debugging stop, does this now mean we're trigging exit twice?"
I added the code to remove file watcher on Close(), or normal exit Close will be called before the fifo is actually deleted.